### ISSUE TITLE
Problems with flag1ad & flag2ad devices

### ISSUE THREAD

**Issue by prjemian** [aps-8id-dys/bluesky#52]
```py
In [1]: listdevice(flag1ad)
Subscription value callback exception (EpicsSignal(read_pv='8idaSoft:flag1:image1:ArrayData', name='flag1ad_image_array_data', parent='flag1ad_image', value=array([], dtype=uint8), timestamp=631152000.0, auto_monitor=True, string=False, write_pv='8idaSoft:flag1:image1:ArrayData', limits=False, put_complete=False))
Traceback (most recent call last):
  File "/home/beams/8IDIUSER/.conda/envs/bstest/lib/python3.10/site-packages/ophyd/device.py", line 325, in __get__
    return instance._signals[self.attr]
KeyError: 'shaped_image'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/beams/8IDIUSER/.conda/envs/bstest/lib/python3.10/site-packages/ophyd/ophydobj.py", line 492, in inner
    cb(*args, **kwargs)
  File "/home/beams/8IDIUSER/.conda/envs/bstest/lib/python3.10/site-packages/ophyd/signal.py", line 696, in _derived_value_callback
    value = self.inverse(value)
  File "/home/beams/8IDIUSER/.conda/envs/bstest/lib/python3.10/site-packages/ophyd/areadetector/base.py", line 123, in inverse
    raise RuntimeError(f"Invalid array size {self.derived_shape}")
RuntimeError: Invalid array size (0, 0, 0)
Subscription value callback exception (EpicsSignal(read_pv='8idaSoft:flag1:image1:ArrayData', name='flag1ad_image_array_data', parent='flag1ad_image', value=array([], dtype=uint8), timestamp=631152000.0, auto_monitor=True, string=False, write_pv='8idaSoft:flag1:image1:ArrayData', limits=False, put_complete=False))
Traceback (most recent call last):
  File "/home/beams/8IDIUSER/.conda/envs/bstest/lib/python3.10/site-packages/ophyd/ophydobj.py", line 492, in inner
    cb(*args, **kwargs)
  File "/home/beams/8IDIUSER/.conda/envs/bstest/lib/python3.10/site-packages/ophyd/signal.py", line 696, in _derived_value_callback
    value = self.inverse(value)
  File "/home/beams/8IDIUSER/.conda/envs/bstest/lib/python3.10/site-packages/ophyd/areadetector/base.py", line 123, in inverse
    raise RuntimeError(f"Invalid array size {self.derived_shape}")
RuntimeError: Invalid array size (0, 0, 0)
RuntimeError: Invalid array size (0, 0, 0)
```

Might be unprimed, might be something else.  Make new issue for this.

_Originally posted by @prjemian in https://github.com/aps-8id-dys/bluesky/issues/51#issuecomment-2299491282_

**Comment by prjemian** [comment]
When this error occurs, the RunEngine starts anyway.  It's a problem from the CA monitor subscription, because no image data has been piped into the Image plugin.

To fix this problem, do these steps in EPICS before starting the bluesky RunEngine:

1. Make sure the area detector Image plugin is enabled.
2. Take at least one image so the dimensions of the data in the Image plugin are not all zeroes.

**Comment by prjemian** [comment]
Nothing to fix in this code.  Wrapping `ad_flag2` object creation with a `try..except` clause.
